docs/usage/inference.rst

```diff
@@ -0,0 +1,168 @@
+    with compiled_net:
+        metrics, callback_data = compiled_net.evaluate(
+            {input_layer: x_test},
+            {output_layer: y_test}
+        )
+
+Parameters:
+    * ``inputs`` - Dictionary mapping input layers to input data
+    * ``targets`` - Dictionary mapping output layers to target data
+    * ``batch_size`` - Optional batch size for processing (default: None)
+    * ``reset_time`` - Whether to reset time between samples (default: True)
+
+Returns:
+    * ``metrics`` - Dictionary containing evaluation metrics
+    * ``callback_data`` - Additional data collected during evaluation
+
+Configuration Options
+-------------------
+The ``InferenceCompiler`` accepts several configuration parameters:
+
+.. code-block:: python
+
+    compiler = InferenceCompiler(
+        example_timesteps=500,    # Number of timesteps per example
+        batch_size=32,           # Default batch size
+        reset_time=True,         # Reset time between samples
+        kernel_profiling=False   # Enable kernel profiling
+    )
+
+Advanced Features
+---------------
+
+Custom Metrics
+~~~~~~~~~~~~
+You can define custom metrics for evaluation:
+
+.. code-block:: python
+
+    def custom_accuracy(output_data, target_data):
+        # Custom metric calculation
+        return accuracy_value
+
+    compiler = InferenceCompiler(metrics={'accuracy': custom_accuracy})
+
+Using Callbacks
+~~~~~~~~~~~~~
+The compiler supports callbacks for monitoring inference:
+
+.. code-block:: python
+
+    class CustomCallback:
+        def on_batch_begin(self, batch):
+            pass
+        
+        def on_batch_end(self, batch):
+            pass
+
+    compiler = InferenceCompiler(callbacks=[CustomCallback()])
+
+Best Practices
+------------
+
+Memory Management
+~~~~~~~~~~~~~~~
+* Use appropriate batch sizes to manage memory usage
+* Clear GPU memory between large inference runs
+* Consider using ``reset_time=False`` for continuous inference
+
+Performance Optimization
+~~~~~~~~~~~~~~~~~~~~~
+* Enable kernel profiling to identify bottlenecks
+* Adjust ``example_timesteps`` based on network dynamics
+* Use appropriate batch sizes for your GPU memory
+
+Common Issues
+-----------
+
+GPU Memory Errors
+~~~~~~~~~~~~~~~
+If encountering GPU memory errors:
+
+* Reduce batch size
+* Clear GPU memory between runs
+* Check for memory leaks in custom callbacks
+
+Performance Issues
+~~~~~~~~~~~~~~~
+If experiencing slow inference:
+
+* Profile kernels to identify bottlenecks
+* Optimize network architecture
+* Adjust simulation timesteps
+
+Complete Example
+--------------
+Here's a complete example showing typical usage:
+
+.. code-block:: python
+
+    from ml_genn.compilers import InferenceCompiler
+
+    # Define custom metric
+    def spike_rate(output_data, target_data):
+        return np.mean(output_data > 0)
+
+    # Create compiler with custom metric
+    compiler = InferenceCompiler(
+        example_timesteps=500,
+        metrics={'spike_rate': spike_rate}
```

Comment:
this is not how metrics are specified. In the example, I would just specify it with a string e.g.
```python
metrics="mean_square_error"
```